fundamentals/minimal-apis/samples/MinApiTestsSample/MinApiTests/IntegrationTests/AuthorizedEndpointsIntegrationTests.cs

```diff
@@ -21,6 +21,7 @@ public AuthorizedEndpointsIntegrationTests(TestWebApplicationFactory<Program> fa
         new object[] { "true", HttpStatusCode.OK }
     };
 
+    // Verify authorized requests can access /admin endpoint.
```

Comment:
Should you add a test to verify antonymous requests fail on this endpoint?

Reply:
I haven't added any authentication for the app. When an anonymous request comes in, it throws a `500 Server Error` saying that the `AddAuthentication()` middleware is not configured. I was thinking that I can add a simple authentication scheme named `proxy`. What do you think?

```
builder.Services.AddAuthentication("ProxyAuth")
    .AddScheme<AuthenticationSchemeOptions, ProxyAuthHandler>("ProxyAuth", null);
```

```
public class ProxyAuthHandler : AuthenticationHandler<AuthenticationSchemeOptions>
{
    public ProxyAuthHandler(IOptionsMonitor<AuthenticationSchemeOptions> options,
        ILoggerFactory logger, UrlEncoder encoder, ISystemClock clock)
        : base(options, logger, encoder, clock)
    {
    }

    protected override Task<AuthenticateResult> HandleAuthenticateAsync()
    {
        var authHeader = Request.Headers["Authorization"].ToString();

        if (authHeader.StartsWith("proxy", StringComparison.OrdinalIgnoreCase))
        {
            var fields = authHeader.Split('_');

            if (fields[0].Equals("proxy") && fields[1].Equals("admin"))
            {
                var adminTicket = CreateAuthTicket(new Claim("admin", "true"));

                return Task.FromResult(AuthenticateResult.Success(adminTicket));
            }

            return Task.FromResult(AuthenticateResult.Fail("Error Parsing Authorization Header"));
        }

        return Task.FromResult(AuthenticateResult.Success(CreateAuthTicket()));
    }

    private static AuthenticationTicket CreateAuthTicket(params Claim[] claims)
    {
        var identity = new ClaimsIdentity(claims, "Proxy");
        var principal = new ClaimsPrincipal(identity);
        return new AuthenticationTicket(principal, "Proxy");
    }
}
```

![image](https://user-images.githubusercontent.com/6568968/192096763-6ee55b99-a0a7-4e27-a7e7-f6efc185bd39.png)

Once it is added, we can add the test cases you mentioned,

```
[Fact]
public async Task GetAdminEndpointReturnsOkForAdminUser()
{
    // Arrange
    var client = _factory.CreateClient();
    client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("proxy_admin");
    
    //Act
    var response = await client.GetAsync("/admin");

    // Assert
    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
}

[Fact]
public async Task GetAdminEndpointReturnsForbiddenForAnonymousUser()
{
    // Arrange
    var client = _factory.CreateClient();

    //Act
    var response = await client.GetAsync("/admin");

    // Assert
    Assert.Equal(HttpStatusCode.Forbidden, response.StatusCode);
}
```

What do you think?

Reply:
@Rick-Anderson never mind the custom authentication scheme. I'll use JWT as it is widely used